src/data/labels.js

```diff
@@ -0,0 +1,326 @@
+        }
+
+        const boardCollection = await boards();
+        const board = await boardCollection.findOne({_id: ObjectId(boardId)});
+        if(board === null){
+            throw new Error('There is not board with that id.');
+        }
+
+        const cardList = board.cards;
+        let foundCard;
+        let returnLabel;
+        
+        if(cardList.length === 0){
+            throw new Error('That board has no cards.');
+        }
+
+        for(let card of cardList){
+            if(card._id.toString() === cardId){
+                foundCard = card;
+            }
+        }
+        if(!foundCard){
+            throw new Error('Card not found');
+        } 
+
+        for(let label of foundCard){
+            if(label._id.toString() === labelId){
+                returnLabel = label;
+            }
+        }
+        if(!returnLabel){
+            throw new Error('Label not found');
+        } 
+        returnLabel._id = returnLabel._id.toString();
+
+        return returnLabel;
+
+    },
+
+    /**
+     * Get the index of a label.
+     * @param {string} boardId The board's id.
+     * @param {string} cardId The card's id.
+     * @param {string} labelId The label's id.
+     * @returns An index of a label.
+     */
+    async getLabelIndexById(boardId, cardId, labelId){
+        if(!boardId || !error_handler.checkObjectId(boardId)){
+            throw new Error('boardId is not valid.');
+        }
+        if(!cardId || !error_handler.checkObjectId(cardId)){
+            throw new Error('cardId is not valid.');
+        }
+        if(!labelId || !error_handler.checkObjectId(labelId)){
+            throw new Error('boardId is not valid.');
+        }
+
+        const boardCollection = await boards();
+        const board = await boardCollection.findOne({_id: ObjectId(boardId)});
+        if(board === null){
+            throw new Error('There is not board with that id.');
+        }
+
+        const cardList = board.cards;
+        let foundCard;
+        let returnLabel;
+        
+        if(cardList.length === 0){
+            throw new Error('That board has no cards.');
+        }
+
+        for(let card of cardList){
+            if(card._id.toString() === cardId){
+                foundCard = card;
+            }
+        }
+        if(!foundCard){
+            throw new Error('Card not found');
+        } 
+
+        let index = 0;
+        for(let label of foundCard){
+            if(label._id.toString() === labelId){
+                returnLabel = label;
+                break;
+            }
+            index++;
+        }
+        if(!returnLabel){
+            throw new Error('Label not found');
+        } 
+
+        return index;
+
+    },
+
+    /**
+     * Get a specific card from a board.
+     * @param {string} boardId The board's id.
+     * @param {string} cardId The card's id.
+     * @returns A card object
+     */
+    async getCardById(boardId, cardId){
```

Comment:
why does this exist in label data?

Reply:
Its used in my other functions, Ill remove this from the exported modules